### ISSUE TITLE
generation of the 'Sec-WebSocket-Key' and 'Mask' for WS header

### ISSUE THREAD

**Issue by smitec11** [frepkovsky/Domoticz-Volumio-Plugin#1]
@Frantisek,  can you check this :

the generation of the 'Sec-WebSocket-Key' header doesnt look right : 
CODE: [SELECT ALL](https://www.domoticz.com/forum/viewtopic.php?t=38358&sid=27aab28b4a715c1bbd1d3f05351dea20#)

def get_mask():
    return str(secrets.randbits(32))

def get_sec_key():
    return str(base64.b64encode(secrets.token_bytes(32)))
should really be
CODE: [SELECT ALL](https://www.domoticz.com/forum/viewtopic.php?t=38358&sid=27aab28b4a715c1bbd1d3f05351dea20#)

def get_mask():
    return secrets.randbits(32)

def get_sec_key():
    return base64.b64encode(secrets.token_bytes(16)).decode("utf-8")
because applying 'str' to bytes data types gives b' at the start of the string

========================
fyi,  the error "Error: (ProcessWholeMessage) Unknown Operation Code (x) encountered", seems to be fixed by Dnpwwo, in the latest beta version of domoticz.

**Comment by frepkovsky** [comment]
@smitec11 - thanks for notifying me. Just for reference I put link to the post on Domoticz forum with recommendation to fix this: 
https://www.domoticz.com/forum/viewtopic.php?p=289389#p289389

**Comment by frepkovsky** [comment]
According to Domoticz source code (hardware/plugins/PluginProtocols.cpp), masking key value for WS protocol  is accepted as 'long' or as 'string' type. 

When returning masking key value from get_mask() as integer (not string), I experienced errors "OverflowError: Python int too large to convert to C long" in Domoticz release 2021.1 on 32-bit OS: see [this post on Domoticz forum](https://www.domoticz.com/forum/viewtopic.php?p=286499#p286499).

In order to keep the plugin compatible with previous and current Domoticz versions (both support masking key value as string type too) I will keep get_mask() code as it is now:

```
def get_mask():
    return str(secrets.randbits(32))
```

Regarding generation of 'Sec-WebSocket-Key' for WS header by function get_sec_key() I will fix it [as recommended](https://www.domoticz.com/forum/viewtopic.php?p=289389#p289389):

```
def get_sec_key():
    return base64.b64encode(secrets.token_bytes(16)).decode("utf-8")
```

**Comment by frepkovsky** [comment]
fixed in version 0.9.6
